Utils/Dataflow/019_esFormat/run.sh

```diff
@@ -3,14 +3,14 @@
 base_dir=$(cd "$(dirname "$(readlink -f "$0")")"; pwd)
 
 usage() {
-  echo "USAGE:
-$(basename "$0") [-h] [-c CONFIG] [--] [ARGS]
+  echo "usage: $(basename "$0") [-h] [-c CONFIG] [--] [ARGS]
```

Comment:
`--` part can not be skipped if `ARGS` is specified, so more accurate way would be something like:
```
   usage: ... [-- [ARGS]]
```

Reply:
I had my doubts about this "argument", but always put them off because something else had to be done. Since we got to this...

First of all, the argument `--` is not explained anywhere (I hope that we'll fix this here, after this discussion). If I understand correctly, it is meant to separate arguments intended for `run.sh` itself from those that it must pass to `esFormat.php`, aka ARGS. However, it seems to be unnecessary, since the [`case`](https://github.com/PanDAWMS/dkb/blob/019-update-action/Utils/Dataflow/019_esFormat/run.sh#L18) processing the arguments has four cases: two for `run.sh`'s arguments, and two others: `--` and `*`, where the latter catches everything that is not the first three. Only difference between `--` and `*` is `shift` for the first, so that `--` itself is not passed to `esFormat.php`.

Therefore, what's the point of the separator if the logic "we only expect `-h` and `-c`, if we encounter something else then it's already ARGS, stop" is in place? One thing I can think of is that this is made in advance, for the situation in the future where `run.sh` would accept more arguments and their amount and type would be arbitrary, hence the need for separator. Another option is possible need to pass something to `esFormat.php` that is similar to `run.sh`'s arguments. Then yes, separator will bring sense into something like `./run.sh -c elastic-search-config -c php-handling-specific-config`.

To summarize, I see 3 options:
1) Implement your suggestion, but also update the logic to demand `--` and document it in `usage()`.
2) Discard your suggestion and simply document `--`, emphasizing the fact that it should only be used to separate aforementioned `-c` and `-c` (in other cases `*` will do just fine).
3) Drop `--` altogether and rely on `*` until (if) the need for the separator arises.

Am I missing something?

Reply:
@Evildoor,
> the argument `--` is not explained anywhere 

Double dash (`--`) is a standard separator for command line arguments (see discussion [here](https://unix.stackexchange.com/questions/11376/what-does-double-dash-mean), for example). Not sure if we need to add some explicit explanation.

>  right now it seems to be unnecessary, since the case processing the arguments has four cases: two for run.sh's arguments, and two others: `--` and `*`, where the latter catches everything that is not the first three. Only difference between `--` and `*` is shift for the first, so that `--` itself is not passed to `esFormat.php`.
>  what's the point of the separator if the logic "we only expect -h and -c, if we encounter something else then it's already ARGS, stop" is in place?

Right. Almost. And no one said that current logic is impeccable, right? ;)
Here's the answer:
1. For now, without `--` you already have no possibility to pass `-h` to `esFormat.php` directly (without running the PHP script itself). No one said you would ever want to, but...
2. Instead of "if it's something unknown (`*`), just go on" there probably should be "if you don't know what it is (`*`), ignore it (do not pass it to `esFormat.php`)".
3. It may also be reasonable to add check for `-*` after `--` to signalize about "unknown option" and exit.

So...
> To summarize, I see 3 options:
> ...
> Am I missing something?

...yes, you're kind of missing the general idea: it's not about "implement/ignore someone's suggestion", it's  more about what do we want to achieve. I'd love to believe the aims here are:
* to finish this PR;
* to make things right.

For the sake of the first aim -- fixing right here every bug detected is not really best way to achieve it (unless the PR was initially about fixing everything).

Talking about the second: `--` separator is required to make sure that user knows what he/she does. In other words, it's here to tell user: "you want to pass some arguments to `esFormat.php` -- say it explicitly". The fact that without `--` it works just the same is nothing but a neglect (mine, actually) that should be questioned and fixed eventually. Now or some time later -- is up to you. I mean, what's gonna take more time: fix it right now or describe the situation in Trello for someone else (or you yourself) who'll have required time later?

---

There's also another thing: why the `run.sh` is used in the first place. Just a thought, but I'd like you to keep it in your mind before deciding what to do next.
The `run.sh` was added, actually, to pass configuration parameters from `es` config file (shaped as Shell variables declaration) to the `PHP` script since it looked easier than do this directly from the PHP (just source the file, no need to parse it and check syntax and all these things required). Should this decision lead to way too many questions now, it may be reasonable to reconsider this design and teach PHP script to read the config directly, without this Shell wrapper. Or not: evaluate the complexity and make your decision :) It's easier to go on the way someone else has chosen, yet there are always reasons to make choice back then and it's better to be aware of the fact that with time and under different circumstances these reasons may look not so persuasive.

Reply:
> > the argument `--` is not explained anywhere
> 
> Double dash (`--`) is a standard separator for command line arguments (see discussion [here](https://unix.stackexchange.com/questions/11376/what-does-double-dash-mean), for example). Not sure if we need to add some explicit explanation.

I see. Still, I think that adding an explanation won't hurt - someone (such as myself at the beginning of this discussion) may be unaware of this standard, and I doubt that 1-2 strings that state the obvious would be much of a problem for the others. Will do.

> ... I'd love to believe the aims here are:
> 
> * to finish this PR;
> * to make things right.
> 
> For the sake of the first aim -- fixing right here every bug detected is not really best way to achieve it (unless the PR was initially about fixing everything).
> 
> Talking about the second: `--` separator is required to make sure that user knows what he/she does. In other words, it's here to tell user: "you want to pass some arguments to `esFormat.php` -- say it explicitly". The fact that without `--` it works just the same is nothing but a neglect (mine, actually) that should be questioned and fixed eventually. Now or some time later -- is up to you. I mean, what's gonna take more time: fix it right now or describe the situation in Trello for someone else (or you yourself) who'll have required time later?

Will look into it.

Reply:
Added explanation for `--`.

Added card for issues with the separator: https://trello.com/c/6vEydzfY. While it is possible that, after some consideration and discussion, the problem would be solved by some small fix, it is also possible that the stage would be redesigned, by removing `run.sh` and moving everything into PHP script, for example. Such (possible) changes do not fit into the idea of "minor fixes" in this PR and, should be left out of it.

Reply:
Ok, thank you!

Reply:
Done, a1eb61b.